### PR TITLE
[R] Enable OpenMP with AppleClang in XGBoost R package

### PR DESCRIPTION
**Pull request by hcho3**
Follow-up to #5232. This PR uses Autoconf to detect the availability of OpenMP library (`libomp`) on Mac OSX. This is done by compiling the following program with command `${CC} -o conftest conftest.c /usr/local/lib/libomp.dylib -Xclang -fopenmp`:

```c
/* conftest.c */
#include <omp.h>

int main(void) {
  return (omp_get_max_threads() <= 1);
    // returns 0 when there's only one thread
    // returns 1 when there's more than one thread
}
```

**Implication: Now Mac users can simply run `install.packages('xgboost')` and enjoy the performance of multiple threads,** as long as they first install `libomp` package from Homebrew. 

I tested this change as follows:
* I ran `R CMD INSTALL .` on my Macbook Pro. A sample XGBoost training job runs with all 4 threads:
```r
library(xgboost)
data(agaricus.train, package='xgboost')
data(agaricus.test, package='xgboost')
train <- agaricus.train
test <- agaricus.test
bst <- xgboost(data = train$data, label = train$label, max.depth = 1, eta = 1,
               nthread = 4, nrounds = 1000, objective = "binary:logistic", verbose = 1,
               eval.metric = 'auc')
```
Screen Shot 2020-01-28 at 12 34 11 AM

* Then I uninstalled `libomp` Homebrew and ran `R CMD INSTALL .` again. A prominent warning shows up, as I intended:
```
checking whether OpenMP will work in a package... no
*****************************************************************************************
WARNING: OpenMP is unavailable on this Mac OSX system. Training speed may be suboptimal.
         To use all CPU cores for training jobs, you should install OpenMP by running

             brew install libomp
*****************************************************************************************
```
Aside from the warning, XGBoost compiles successfully. The sample job ran without a problem, albeit this time with a single thread.

Note to reviewer: you should only review `configure.ac`. The `configure` script is generated from `configure.ac` when `autoconf` is run.

### CODE REVIEW

File: R-package/configure.ac [hunk 1]
```diff
@@ -4,6 +4,9 @@ AC_PREREQ(2.62)
 
 AC_INIT([xgboost],[0.6-3],[],[xgboost],[])
 
+# Use this line to set CC variable to a C compiler
+AC_PROG_CC
```
Comment (trivialfis) [inline R-package/configure.ac]: ?  Why ?

Comment (hcho3) [reply R-package/configure.ac]: The feature testing script needs access to `CC` variable.

Comment (trivialfis) [reply R-package/configure.ac]: @hcho3 Yup, but why setting it to a `C` complier?  I assume that means changing `g++` to `gcc` or `clang++` to `clang`.

Comment (hcho3) [reply R-package/configure.ac]: `CC` refers to a C compiler and `CXX` refers to a C++ compiler.

Comment (trivialfis) [reply R-package/configure.ac]: Yup.   I thought we use `CXX` var exclusively so are compilation tests ... Never mind,  I don't know anything about autotools.  Sorry for the noise.

Comment (hcho3) [reply R-package/configure.ac]: Don't worry, I don't know much about autotools either. I wait for the day where we can use CMake for CRAN submissions.

**Pull request merged**